File: benchmark/c/cpw/sum_series_23_45_67_89_upto_n_terms/sum_series_23_45_67_89_upto_n_terms.c


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>



float sum_series_23_45_67_89_upto_n_terms ( int n ) ;
float sum_series_23_45_67_89_upto_n_terms ( int n ) {
  int i = 1;
  double res = 0.0;
  bool sign = 1;
  while ( n > 0 ) {
    n --;
    if ( sign ) {
      sign = ! sign;
      res = res + ( double ) ++ i / ++ i;
    }
    else {
      sign = ! sign;
      res = res - ( double ) ++ i / ++ i;
    }
  }
  return res;
}

File: translation_result/else_bms/success/c/cpw/sum_series_23_45_67_89_upto_n_terms/gemini/BaseRepair/sum_series_23_45_67_89_upto_n_terms.rs

fn sum_series_23_45_67_89_upto_n_terms(mut n: i32) -> f64 {
    let mut i = 1;
    let mut res = 0.0;
    let mut sign = true;
    while n > 0 {
        n -= 1;
        if sign {
            sign = false;
            res += (i + 1) as f64 / (i + 1) as f64;
        } else {
            sign = true;
            res -= (i + 1) as f64 / (i + 1) as f64;
        }
        i += 1;
    }
    res
}
